internal/langserver/handlers/hooks_module.go

```diff
@@ -153,7 +153,7 @@ func updateDiagnostics(ctx context.Context, notifier *diagnostics.Notifier) stat
 	}
 }
 
-func refreshModuleProviders(ctx context.Context, clientRequester session.ClientCaller, logger *log.Logger, commandId string) state.ModuleChangeHook {
+func callClientCommand(ctx context.Context, clientRequester session.ClientCaller, logger *log.Logger, commandId string) state.ModuleChangeHook {
 	return func(oldMod, newMod *state.Module) {
 		_, err := clientRequester.Callback(ctx, commandId, nil)
 		if err != nil {
```

Comment:
So the method should return early is either `oldMod` or `newMode` is nil?

Reply:
I think we still want to reflect both removal or introduction in the UI, so we should still refresh, but we shouldn't crash upon logging just because it's removal.

Maybe something like this would help?

```go
var modPath string
if oldMod != nil {
  modPath = oldMod.Path
} else {
  modPath = newMod.Path
}

```

Reply:
Ah, now I understand. Modified!